packages/CodingStandard/src/Fixer/Order/MethodOrderByTypeFixer.php

```diff
@@ -0,0 +1,267 @@
+
+            if ($sorted !== $methodElements) {
+                $this->privatesCaller->callPrivateMethod(
+                    $this->orderedClassElementsFixer,
+                    'sortTokens',
+                    $tokens,
+                    $startIndex,
+                    $endIndex,
+                    $sorted
+                );
+            }
+        }
+    }
+
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    public function getName(): string
+    {
+        return self::class;
+    }
+
+    public function isRisky(): bool
+    {
+        return false;
+    }
+
+    public function supports(SplFileInfo $file): bool
+    {
+        return true;
+    }
+
+    /**
+     * @param mixed[]|null $configuration
+     */
+    public function configure(?array $configuration = null): void
+    {
+        if ($configuration === null) {
+            return;
+        }
+
+        $this->configuration = $this->getConfigurationDefinition()
+            ->resolve($configuration);
+    }
+
+    public function getConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        $fixerOptionBuilder = new FixerOptionBuilder(self::METHOD_ORDER_BY_TYPE_OPTION, 'Methods order by type.');
+
+        $methodsOrderByTypeOption = $fixerOptionBuilder->setAllowedTypes(['array'])
+            ->setDefault([])
+            ->getOption();
+
+        return new FixerConfigurationResolver([$methodsOrderByTypeOption]);
+    }
+
+    /**
+     * @param mixed[] $elements
+     * @return mixed[]
+     */
+    private function filterPublicMethodElementsFirst(array $elements): array
+    {
+        $publicMethods = [];
+        $restOfMethods = [];
+
+        foreach ($elements as $element) {
+            if ($element['type'] !== 'method') {
+//                $restOfMethods[$element['name']] = $element;
+                continue;
+            }
+
+            if ($element['visibility'] === 'public') {
+                $publicMethods[$element['name']] = $element;
+            } else {
+                $restOfMethods[$element['name']] = $element;
+            }
+//            $publicMethods[$element['name']] = $element;
+        }
+
+        return array_merge($publicMethods, $restOfMethods);
+    }
+
+    private function matchClassType(ClassWrapper $classWrapper): ?string
+    {
+        /** @var string $type */
+        foreach (array_keys($this->configuration[self::METHOD_ORDER_BY_TYPE_OPTION]) as $type) {
+            if ($this->isClassWrapperByParentType($classWrapper, $type)) {
+                return $type;
+            }
+        }
+
+        return null;
+    }
+
+    private function isClassWrapperByParentType(ClassWrapper $classWrapper, string $type): bool
+    {
+        if ($classWrapper->getParentClassName()) {
+            if ($classWrapper->getParentClassName() === $type) {
+                return true;
+            }
+        }
+
+        foreach ($classWrapper->getInterfaceNames() as $interfaceName) {
```

Comment:
Thanks in_array

Reply:
I do need to finish that Rector Code Quality level :cry: